src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -1964,6 +1964,14 @@ public final void clearBatch() throws SQLServerException {
                 ArrayList<String> columnList = parseUserSQLForColumnListDW();
                 ArrayList<String> valueList = parseUserSQLForValueListDW(false);
 
+                for (int i = 0; i < valueList.size(); i++) {
+                    if (!valueList.get(i).equals("?")) {
+                        // throw IllegalArgumentException and fallback to original logic for batch insert
+                        throw new IllegalArgumentException(
+                                "Only fully parameterized queries are allowed for using Bulk Copy API for batch insert at the moment.");
```

Comment:
Why don't we fallback to batch insert Original implementation instead of throwing exception here? It may not be a query issue, its just the 'useBulkCopyWithBatchInserts' API will not parse partial parameterized queries. Applies to all other exceptions.